website/versioned_docs/version-0.63/native-modules-android.md

```diff
@@ -445,29 +640,33 @@ public class ImagePickerModule extends ReactContextBaseJavaModule {
 }
 ```
 
-### Listening to LifeCycle events
+### Listening to Lifecycle Events
 
-Listening to the activity's LifeCycle events such as `onResume`, `onPause` etc. is very similar to how we implemented `ActivityEventListener`. The module must implement `LifecycleEventListener`. Then, you need to register a listener in the module's constructor,
+Listening to the activity's LifeCycle events such as `onResume`, `onPause` etc. is very similar to how we implemented `ActivityEventListener`. The module must implement `LifecycleEventListener`. Then, you need to register a listener in the module's constructor like so:
 
-```java
+```jsx
 reactContext.addLifecycleEventListener(this);
 ```
 
 Now you can listen to the activity's LifeCycle events by implementing the following methods:
 
-```java
+```jsx
 @Override
 public void onHostResume() {
-    // Activity `onResume`
+   // Activity `onResume`
 }
-
 @Override
 public void onHostPause() {
-    // Activity `onPause`
+   // Activity `onPause`
 }
-
 @Override
 public void onHostDestroy() {
-    // Activity `onDestroy`
+   // Activity `onDestroy`
 }
 ```
+
+### Threading
+
+To date, on Android, all native module async methods execute on one thread. On iOS, each native module can provide its method queue, or have React Native generate a method queue for it. On iOS, native modules also have an API to get a hold of the method queue on which their async methods are executed.
+
+Native modules should not have any assumptions about what thread they are being called on, as the current assignment is subject to change in the future. If a blocking call is required, the heavy work should be dispatched to an internally managed worker thread, and any callbacks distributed from there.
```

Comment:
It looks like we don't mention the `invalidate` Android analogue. In Java, NativeModules can have a `onCatalystInstanceDestroy()` method that they can implement to do cleanup of any resources they own. There's also an initialize method: https://github.com/facebook/react-native/blob/ae418b83c071d27d1a5bc7dcb46d546686bd21b8/ReactAndroid/src/main/java/com/facebook/react/bridge/NativeModule.java#L40-L51.

Reply:
@RSNara I ended up removing the notes about iOS here. Do you want to add this info about Android here?